File: paper/src/main/java/com/al3x/housing2/Data/TeamData.java
package com.al3x.housing2.Data;

import com.al3x.housing2.Instances.Team;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class TeamData {
    private String name;
    private String prefix;
    private String color;
    private String displayName;
    private String suffix;
    private boolean friendlyFire;

    public TeamData() {
    }

    public TeamData(String name, String prefix, String color, String displayName, String suffix, boolean friendlyFire) {
        this.name = name;
        this.prefix = prefix;
        this.color = color;
        this.displayName = displayName;
        this.suffix = suffix;
        this.friendlyFire = friendlyFire;
    }


    public static List<TeamData> fromList(List<Team> list) {
        List<TeamData> teamData = new ArrayList<>();
        for (Team team : list) {
            teamData.add(team.toData());
        }
        return teamData;
    }

    public static List<Team> toList(List<TeamData> data) {
        List<Team> teams = new ArrayList<>();
        for (TeamData teamData : data) {
            teams.add(new Team(teamData));
        }
        return teams;
    }
}

File: src/main/java/com/al3x/housing2/Instances/HousingData/TeamData.kt
package com.al3x.housing2.Instances.HousingData

import com.al3x.housing2.Instances.Team

data class TeamData(
    var name: String,
    var prefix: String,
    var color: String,
    var displayName: String,
    var suffix: String,
    var friendlyFire: Boolean,
) {
    companion object {
        fun fromList(list: List<Team>): List<TeamData> {
            val teamData = mutableListOf<TeamData>()
            list.forEach {
                teamData.add(it.toData())
            }
            return teamData
        }

        fun toList(data: List<TeamData>): List<Team> {
            val teams = mutableListOf<Team>()
            data.forEach {
                teams.add(Team(it))
            }
            return teams
        }
    }
}